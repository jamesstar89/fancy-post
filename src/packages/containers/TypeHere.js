import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { typeHereAction } from '../redux/actions';
import { TypeHereContainer, TextFieldEntry } from './styles/styled';

export class TypeHere extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fontSize: '40px',
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    const { value } = e.target;
    if (value.length > 3) {
      this.props.typeHereAction(value);
    }
    if (value.length > 7) {
      this.setState({
        fontSize: '35px'
      });
    }
  }

  render() {
    const { preview, previewAddons } = this.props;
    const Preview = preview;

    return (
      <TypeHereContainer>
        <TextFieldEntry
          type="text"
          element="textarea"
          placeholder="Type something"
          minLength={2}
          debounceTimeout={300}
          style={{
            fontSize: this.state.fontSize,
            lineHeight: this.state.fontSize,
          }}
          onChange={this.onChange}
        />
        <Preview previewAddons={previewAddons} />
      </TypeHereContainer>
    );
  }
}

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    typeHereAction,
  },
  dispatch
);

export default connect(undefined, mapDispatchToProps)(TypeHere);
