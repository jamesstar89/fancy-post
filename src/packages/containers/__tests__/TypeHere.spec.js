import React from 'react';
import { mount } from 'enzyme';
import TypeHere from '../TypeHere';

describe('TypeHere', () => {
  test('Should render', () => {
    const Preview = () => <div />;
    const tree = mount(<TypeHere preview={Preview} />);
    expect(tree).toMatchSnapshot();
  });
});
