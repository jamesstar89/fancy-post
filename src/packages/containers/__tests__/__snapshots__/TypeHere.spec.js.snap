// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeHere Should render 1`] = `
<TypeHere
  preview={[Function]}
>
  <ForwardRef>
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "ghuPdm",
            "rules": Array [
              "
  position: relative;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bwzfXH ghuPdm"
      >
        <ForwardRef
          debounceTimeout={300}
          element="textarea"
          minLength={2}
          onChange={[Function]}
          placeholder="Type something"
          style={
            Object {
              "fontSize": "40px",
              "lineHeight": "40px",
            }
          }
          type="text"
        >
          <StyledComponent
            debounceTimeout={300}
            element="textarea"
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "DebounceInput": [Function],
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "pntFT",
                  "rules": Array [
                    "
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  -o-appearance: none;
  appearance: none;
  border: 0;
  background: none;
  color: white;
  outline: 0;
  width: calc(100% - 15px);
  overflow: hidden;
  resize: none;
  position: absolute;
  top: 10px;
  left: 15px;
  z-index: 10;
  height: 280px;
  line-height: 70px;
  -webkit-transition: all 1s;
  transition: all 1s;

  &::placeholder {
    color: rgba(255, 255, 255, 0.2);
  }
",
                  ],
                },
                "displayName": "Styled(DebounceInput)",
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            forwardedRef={null}
            minLength={2}
            onChange={[Function]}
            placeholder="Type something"
            style={
              Object {
                "fontSize": "40px",
                "lineHeight": "40px",
              }
            }
            type="text"
          >
            <DebounceInput
              className="sc-htpNat pntFT"
              debounceTimeout={300}
              element="textarea"
              forceNotifyByEnter={true}
              forceNotifyOnBlur={true}
              minLength={2}
              onChange={[Function]}
              placeholder="Type something"
              style={
                Object {
                  "fontSize": "40px",
                  "lineHeight": "40px",
                }
              }
              type="text"
            >
              <textarea
                className="sc-htpNat pntFT"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Type something"
                style={
                  Object {
                    "fontSize": "40px",
                    "lineHeight": "40px",
                  }
                }
                type="text"
                value=""
              />
            </DebounceInput>
          </StyledComponent>
        </ForwardRef>
        <Preview>
          <div />
        </Preview>
      </div>
    </StyledComponent>
  </ForwardRef>
</TypeHere>
`;
